src/Sarif.Sarifer/IBackgroundAnalyzer.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    /// <summary>
+    /// Interface for performing a single static analysis in the background.
+    /// </summary>
+    public interface IBackgroundAnalyzer
+    {
+        /// <summary>
+        /// Performs a single analysis on the specified text.
+        /// </summary>
+        /// <param name="text">
+        /// The text to analyze
+        /// </param>
+        void StartAnalysis(string path, string text);
```

Comment:
>StartAnalysis [](start = 13, length = 13)

We should probably be consistent and have a stream version as well

Reply:
I don't think so. Remember that the `path` here is the path to a source file we just opened in the editor (or, per @michaelfanning's ultimate goal, one of the source files in the solution that you just right-clicked and selected "Perform [redacted] analysis"). We won't be streaming content into this API.

---
In reply to: [519920052](https://github.com/microsoft/sarif-visualstudio-extension/pull/260#discussion_r519920052) [](ancestors = 519920052)